### PR TITLE
pre-assign token hash seeds

### PR DESCRIPTION
**Pull request by ryley-o**
Document steps to pre-assign token hash seeds on shared randomizer

linear: https://linear.app/art-blocks/issue/PLT-554/document-pre-assigned-hash-seed-steps

### CODE REVIEW

File: minter-suite/minter-suite-supplemental.md [hunk 1]
```diff
@@ -139,3 +139,49 @@ console.log(
...
+
+   - `coreContract`: The address of the core contract for the project
+   - `tokenId`: The token ID to check the hash seed for
+
+1. MINT! Ordering may be critical if minting specific tokens to specific addresses, so the artist may need to keep the project paused and call purchaseTo in a specific order. No other changes to the minting process are required. Confirming token hashes are as expected along the way is highly recommended.
+
+1. After minting is complete, the artist may call `toggleProjectUseAssignedHashSeed()` to disable the use of assigned hash seeds for the project, and allow the randomizer to pseudorandomly assign hash seeds to tokens as normal. Additionally, the hash seed setter contract may be set back to the zero address by calling `setHashSeedSetterContract` from the artist wallet.
+
+!!!info
+The steps above follow the pattern implemented in the following test case: [shared-randomizer-v0/configure.test.ts:assigns_expected_token_when_polyptych"](https://github.com/ArtBlocks/artblocks-contracts/blob/1d2a6ad5915cb225c6b0d31710568a439b0e212b/packages/contracts/test/randomizer/shared-randomizer-v0/configure.test.ts#L269)
```
Comment (aaronpenne) [inline minter-suite/minter-suite-supplemental.md:186]: love to see the ref

File: minter-suite/minter-suite-supplemental.md [hunk 2]
```diff
@@ -139,3 +139,49 @@ console.log(
...
+   - `hashSeed`: The hash seed to assign to the token
+
+1. You may confirm that token hash seeds have been properly assigned by calling the `preAssignedHashSeed()` read-only function on `SharedRandomizerV0` with the following parameters:
+
+   - `coreContract`: The address of the core contract for the project
+   - `tokenId`: The token ID to check the hash seed for
+
+1. MINT! Ordering may be critical if minting specific tokens to specific addresses, so the artist may need to keep the project paused and call purchaseTo in a specific order. No other changes to the minting process are required. Confirming token hashes are as expected along the way is highly recommended.
+
+1. After minting is complete, the artist may call `toggleProjectUseAssignedHashSeed()` to disable the use of assigned hash seeds for the project, and allow the randomizer to pseudorandomly assign hash seeds to tokens as normal. Additionally, the hash seed setter contract may be set back to the zero address by calling `setHashSeedSetterContract` from the artist wallet.
```
Comment (aaronpenne) [inline minter-suite/minter-suite-supplemental.md:183]: imo we should reduce any ambiguity by being explicit and direct about what to do, for example changing `Additionally, the hash seed setter contract may be set back to the zero address...` to `<new step or substep>Set the hash seed setter contract to the zero address...`

File: minter-suite/minter-suite-supplemental.md [hunk 3]
```diff
@@ -139,3 +139,49 @@ console.log(
...
+
+## Pre-Assigning Token Hash Seeds
+
+The shared randomizer `SharedRandomizerV0` enables an artist to pre-assign token hash seeds for tokens on a project. This allows an artist to assign a specific hash seed to a token before it is minted, and then mint the token with that hash seed at a later time.
+
+The process for pre-assigning token hash seeds is as follows:
+
+1. Identify and confirm that your contract is using a `SharedRandomizerV0` contract. Visit your core contract on etherscan (or appropriate L2 explorer) and inspect the read-only function `randomizerContract()` to confirm that the randomizer contract is a `SharedRandomizerV0`.
+
+1. Navigate to the `SharedRandomizerV0` contract on etherscan (or appropriate L2 explorer), and connect your artist wallet. Alternatively, if your artist wallet is a multisig, interact with the `sharedRandomizerV0` in something like Gnosis Safe's Transaction Builder (this is more efficient and convenient when assigning hash seeds for many tokens).
```
Comment (aaronpenne) [inline minter-suite/minter-suite-supplemental.md:153]: `(or appropriate L2 explorer)` -> `(or appropriate L2 explorer, eg. basescan)`

File: minter-suite/minter-suite-supplemental.md [hunk 4]
```diff
@@ -139,3 +139,49 @@ console.log(
   merkleProof
 );
 ```
+
+---
+
+## Pre-Assigning Token Hash Seeds
+
+The shared randomizer `SharedRandomizerV0` enables an artist to pre-assign token hash seeds for tokens on a project. This allows an artist to assign a specific hash seed to a token before it is minted, and then mint the token with that hash seed at a later time.
```
Comment (aaronpenne) [inline minter-suite/minter-suite-supplemental.md:147]: would be nice in an info callout or some other highlight

**Review by aaronpenne — APPROVED** [review]
couple nits but none blocking, lgtm

File: minter-suite/minter-suite-supplemental.md (hunk 2 above)
Comment (ryley-o) [reply minter-suite/minter-suite-supplemental.md]: updated in **[977abdc](https://github.com/ArtBlocks/artblocks-docs/pull/312/commits/977abdc26fbfcd51f847cc118c13ce7bac59b0fa)**

File: minter-suite/minter-suite-supplemental.md (hunk 3 above)
Comment (ryley-o) [reply minter-suite/minter-suite-supplemental.md]: updated in [977abdc](https://github.com/ArtBlocks/artblocks-docs/pull/312/commits/977abdc26fbfcd51f847cc118c13ce7bac59b0fa)

File: minter-suite/minter-suite-supplemental.md (hunk 4 above)
Comment (ryley-o) [reply minter-suite/minter-suite-supplemental.md:148]: updated in [977abdc](https://github.com/ArtBlocks/artblocks-docs/pull/312/commits/977abdc26fbfcd51f847cc118c13ce7bac59b0fa)

**Pull request merged**